### ISSUE TITLE
[question] How to access a value in the store from 'app/utils/request.js'?

### ISSUE THREAD

**Issue by elyran** [mxstbr/react-boilerplate#1470]
Hi,

First, thank you for the boilerplate! You definitely put much effort into it and it's much appreciated :)

**I am trying to add a certain variable to every HTTP request make through 'app/utils/request.js'.
This variable is stored at my store.**

I'm trying to think about the best way to access this variable, any suggestion?

Current idea:
Have a module named 'app/store-globals.js'.
It's a singleton which accepts an initial 'store' object, and has a 'getValue()' method, encapsulating the store while providing read-only access to other modules.

**Comment by gihrig** [comment]
In the spirit of redux, I think you would want to do this data transformation closer to the store, e.g. in the saga(s) that make calls to the request helper. The sagas have access to the store, request does not.

Issue #1277 is probably a good place to start.

Similar issues have been discussed, for different reasons and several solutions have been offered in these: #1384, #1315, #1420 and #1447. Perhaps you can get some ideas and background from these.

**Comment by elyran** [comment]
@gihrig I agree, however, it won't be enough in my case.

I have many API calls I make from different places, so I don't want to add this parameter to each one.
I would like to have the parameter added automatically.

Two options:

1. Create a Redux middleware which adds the parameter to the action. Not a clean solution, pollutes every action (not related to API calls) with redundant data.

2. Create a global saga which has priority (because I set it up early on). When it gets the 'CALL_API' action, it does:
  a. Add the parameter (an auth token in my case).
  b. Call the API, using 'fetch()' as in 'app/utils/request.js'.
  c. Return the response.

At the moment I'm trying to figure out how to return the response.
In other words, and it's more of a Saga question - do you know how to use the 'put' effect which triggers an action (in my case that would be making an API request) and then return a value, like the 'call' effect does?

**Comment by KarandikarMihir** [comment]
@elyran Does this help? [redux-optimist](https://github.com/ForbesLindesay/redux-optimist)

**Comment by elyran** [comment]
@KarandikarMihir I don't think it's what I'm looking for. Thanks though

**Comment by KarandikarMihir** [comment]
No problem!

**Comment by elyran** [comment]
> Create a Redux middleware which adds the parameter to the action. Not a clean solution, pollutes every action (not related to API calls) with redundant data.

Regarding this one - will not work. Because I call (using 'call') and expect a reply so the saga continue.

So far the initial idea/approach works, although not very Redux'ish.. Suggestions are welcome.
